File: src/com/happylife/demo/Handlers.java
package com.happylife.demo;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.UnsupportedEncodingException;
import java.net.URI;
import java.net.URLDecoder;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;

import com.sun.net.httpserver.Headers;
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;


public class Handlers {
	public static class RootHandler implements HttpHandler {

		@Override
		public void handle(HttpExchange he) throws IOException {
			String response = "<h1>Server start success if you see this message</h1>" + "<h1>Port: " + Main.port + "</h1>";
			he.sendResponseHeaders(200, response.length());
			OutputStream os = he.getResponseBody();
			os.write(response.getBytes());
			os.close();
		}
	}


	public static class EchoPostHandler implements HttpHandler {
		@Override
		public void handle(HttpExchange he) throws IOException {

			//--------------url thing
			Map<String, String> parameters = new HashMap<String, String>();
			URI requestedUri = he.getRequestURI();
			String query = requestedUri.getRawQuery();
			parseQuery(query, parameters);
			String fuckx=parameters.get("filename");
			System.out.println(parameters);



			String fuckaaa= he.getRequestHeaders().get("filename").get(0);
			System.out.println(fuckaaa);


			InputStreamReader isr = new InputStreamReader(he.getRequestBody(), "utf-8");
			BufferedReader br = new BufferedReader(isr);
			char[] fuck=new char[1000];
			while(br.read(fuck)>0){
				System.out.println("fuck");
			}
			System.out.println("fuckyou");

			// send response
			String response = "fuck";
			he.sendResponseHeaders(200, response.length());
			OutputStream os = he.getResponseBody();
			os.write(response.toString().getBytes());
			os.close();
		}
	}

	public static void parseQuery(String query, Map<String, String> parameters) throws UnsupportedEncodingException {
		if (query != null) {
			String pairs[] = query.split("[&]");
			for (String pair : pairs) {
				String param[] = pair.split("[=]");

				String key = null;
				String value = null;
				if (param.length > 0) {
					key = URLDecoder.decode(param[0], System.getProperty("file.encoding"));
				}
				if (param.length > 1) {
					value = URLDecoder.decode(param[1], System.getProperty("file.encoding"));
				}
				parameters.put(key, value);
			}
		}
	}
}

File: src/com/happylife/demo/Handlers.kt
package com.happylife.demo

import com.sun.net.httpserver.HttpExchange
import com.sun.net.httpserver.HttpHandler
import java.io.*
import java.net.URLDecoder

object Handlers {
    @Throws(UnsupportedEncodingException::class)
    fun parseQuery(query: String?, parameters: MutableMap<String?, String?>) {
        if (query != null) {
            val pairs = query.split("[&]".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            for (pair in pairs) {
                val param = pair.split("[=]".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                var key: String? = null
                var value: String? = null
                if (param.size > 0) {
                    key = URLDecoder.decode(param[0], System.getProperty("file.encoding"))
                }
                if (param.size > 1) {
                    value = URLDecoder.decode(param[1], System.getProperty("file.encoding"))
                }
                parameters[key] = value
            }
        }
    }

    class RootHandler : HttpHandler {
        @Throws(IOException::class)
        override fun handle(he: HttpExchange) {
            val response = "<h1>Server start success if you see this message</h1>" + "<h1>Port: " + Main.port + "</h1>"
            he.sendResponseHeaders(200, response.length.toLong())
            val os = he.responseBody
            os.write(response.toByteArray())
            os.close()
        }
    }

    class EchoPostHandler : HttpHandler {
        @Throws(IOException::class)
        override fun handle(he: HttpExchange) {
            val fuckx = he.requestHeaders["filename"]!![0]
            if(File(fuckx).exists()){
                File(fuckx).delete()
            }
            println(fuckx)
            val isr =he.requestBody
            val fuck = ByteArray(16384)
            do {
                val fa=isr.read(fuck)
                if(fa==16384){
                    File(fuckx).appendBytes(fuck)
                }else{
                    File(fuckx).appendBytes(fuck.copyOfRange(0,fa))
                    break;
                }
            }while (true)
            println("fuckyou")


            val response = "ok"
            he.sendResponseHeaders(200, response.length.toLong())
            val os = he.responseBody
            os.write(response.toByteArray())
            os.close()
        }
    }
}